clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -3307,6 +3307,29 @@ void Sema::CheckShadowInheritedFields(const SourceLocation &Loc,
   }
 }
 
+template <typename AttrType>
+inline static bool HasAttribute(const QualType &T) {
+  if (const TagDecl *TD = T->getAsTagDecl())
+    return TD->hasAttr<AttrType>();
+  if (const TypedefType *TDT = T->getAs<TypedefType>())
+    return TDT->getDecl()->hasAttr<AttrType>();
+  return false;
+}
+
+inline static bool IsUnusedPrivateField(FieldDecl *FD) {
```

Comment:
This function doesn't actually check whether it is unused.  So the name here is really confusing/misleading.

Reply:
@erichkeane I was thinking of using `IsPrivateFieldWithoutSideEffects` to proceed with adding it to `UnusedPrivateFields`, however, decided to match with `UnusedPrivateFields`...  What do you think about this?

Reply:
I again am not sure I get what you're suggesting?  I'm sorry, I don't get the proposal.  Something like `IsPotentiallyUnusedPrivateField` or something might work, but the fact that it is not actually checking if it is used is problematic.